hazelcast/src/main/java/com/hazelcast/sql/impl/exec/AbstractExec.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec;
+
+import com.hazelcast.sql.impl.fragment.QueryFragmentContext;
+import com.hazelcast.sql.impl.row.EmptyRowBatch;
+import com.hazelcast.sql.impl.row.RowBatch;
+
+/**
+ * Abstract executor.
+ */
+public abstract class AbstractExec implements Exec {
+
+    protected QueryFragmentContext ctx;
+    private final int id;
+
+    protected AbstractExec(int id) {
+        this.id = id;
+    }
+
+    @Override
+    public int getId() {
+        return id;
+    }
+
+    @Override
+    public final void setup(QueryFragmentContext ctx) {
+        this.ctx = ctx;
+
+        setup0(ctx);
+    }
+
+    @Override
+    public final IterationResult advance() {
+        return advance0();
+    }
+
+    @Override
+    public final RowBatch currentBatch() {
+        RowBatch res = currentBatch0();
+
+        return res != null ? res : EmptyRowBatch.INSTANCE;
+    }
+
+    protected void setup0(QueryFragmentContext ctx) {
+        // No-op.
```

Comment:
If it's a no-op method, we can make it abstract.

Reply:
I made it non-absrtact, because some operators do not require setup. So instead of having multiple no-op implementations in operators, we have only one.